From fd666fc2bd5f7aacc997a7859e3fb17ad3de5c83 Mon Sep 17 00:00:00 2001
From: Bryan Quigley <code@bryanquigley.com>
Date: Wed, 30 Dec 2020 11:39:37 -0800
Subject: [PATCH] This just brings the debian patch to git from   [ Michael
 Vogt ]   * gtk: fix invalid assignment from SizeToStr to intermediate value  
   (Closes: 974790, 974725)

---
 gtk/gtkpkglist.cc | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/gtk/gtkpkglist.cc b/gtk/gtkpkglist.cc
index 18d9078d..0363b9e2 100644
--- a/gtk/gtkpkglist.cc
+++ b/gtk/gtkpkglist.cc
@@ -404,6 +404,7 @@ gtk_pkg_list_get_value(GtkTreeModel *tree_model,
       return;
    }
 
+   std::string cppstr;
    const gchar *str;
    switch (column) {
       case NAME_COLUMN:
@@ -412,13 +413,13 @@ gtk_pkg_list_get_value(GtkTreeModel *tree_model,
          break;
       case PKG_SIZE_COLUMN:
          if (pkg->installedVersion()) {
-            str = SizeToStr(pkg->installedSize()).c_str();
-            g_value_set_string(value, str);
+            cppstr = SizeToStr(pkg->installedSize());
+            g_value_set_string(value, cppstr.c_str());
          }
          break;
       case PKG_DOWNLOAD_SIZE_COLUMN:
-	 str = SizeToStr(pkg->availablePackageSize()).c_str();
-	 g_value_set_string(value, str);
+	 cppstr = SizeToStr(pkg->availablePackageSize());
+	 g_value_set_string(value, cppstr.c_str());
          break;
       case SECTION_COLUMN:
 	 str = pkg->section();
